spec.html

```diff
@@ -835,7 +833,10 @@ <h1>ToElementDescriptor ( _elementObject_ )</h1>
         1. Otherwise,
           1. Set _descriptor_ to ? ToPropertyDescriptor(_descriptor_).
         1. Let _initializer_ be ? Get(_elementObject_, `"initializer"`).
+        1. If _kind_ is `"initializer"` and _initializer_ is *undefined*, throw a *TypeError* exception
         1. Let _elements_ be ? Get(_elementObject_, `"elements"`).
+        1. If _kind_ is `"initializer"`,
+          1. If _key_ is not *undefined*, throw a *TypeError* exception.
```

Comment:
shouldn't this check happen right after `Let _key_ be ? Get(_elementObject_, `"key"`)`?

Reply:
I previously tried to sort the validations towards the end, but I can do them mixed inline if you prefer.

Reply:
I feel like in general we strive to have validations occur as soon as possible, to avoid unnecessary observable behavior.